inc/class-wpseo-utils.php

```diff
@@ -191,34 +191,35 @@ public static function trim_recursive( $value ) {
 	 * @return string
 	 */
 	public static function translate_score( $val, $css_value = true ) {
-		if ( $val > 10 ) {
-			$val = round( $val / 10 );
-		}
-		switch ( $val ) {
-			case 0:
+
+		switch ( true ) {
+
+			case 0 === $val:
 				$score = __( 'N/A', 'wordpress-seo' );
-				$css   = 'na';
+				$css   = WPSEO_Rank::NO_FOCUS;
+				break;
+
+			default:
+			case WPSEO_Rank::$range[ WPSEO_Rank::BAD ]['start'] <= $val && $val <= WPSEO_Rank::$range[ WPSEO_Rank::BAD ]['end']:
+				$score = __( 'Bad', 'wordpress-seo' );
+				$css   = WPSEO_Rank::BAD;
 				break;
-			case 4:
-			case 5:
+
+			case WPSEO_Rank::$range[ WPSEO_Rank::POOR ]['start'] <= $val && $val <= WPSEO_Rank::$range[ WPSEO_Rank::POOR ]['end']:
```

Comment:
Why not determine the score by looping over the `WPSEO_Rank::$range` array. Just breaking the loop on the score that will match. This gives the power to add new ranks and keeps the logic in the `WPSEO_Rank` class.